spec/unit/puppet-strings/yard/handlers/ruby/provider_handler_spec.rb

```diff
@@ -27,6 +27,34 @@
     end
   end
 
+  describe 'parsing a provider with an invalid docstring assignment' do
+    let(:source) { <<-SOURCE
+Puppet::Type.type(:custom).provide :linux do
+  @doc = 123
```

Comment:
@er0ck I think you *can* reference the @doc instance var, but the only record of this being a common thing I could find was https://projects.puppetlabs.com/issues/11404.

I added the test to cover this bit of strings code: https://github.com/puppetlabs/puppet-strings/blob/master/lib/puppet-strings/yard/handlers/ruby/provider_handler.rb#L52-L55